pkg/resources/homeResource.go

```diff
@@ -19,7 +19,11 @@ func (hsh HomeServiceHandler) ServeHTTP(w http.ResponseWriter, r *http.Request)
 	isAuthorized, shelterSession := hsh.isAuthorized(r)
 
 	if !isAuthorized {
+		tpl, _ := retrievers.RetrieveTemplate("home/unauthorized")
```

Comment:
I'm wondering here that is we don't have a session, we redirect to the login page and if we do have a session, then we show the unauthorized page.